### PR TITLE
feat(oxc_cli): support no-ignore

### LINKED ISSUE #54: Feat(cli): Ignore files
https://eslint.org/docs/latest/use/command-line-interface#ignore-files

It would be the best if you submit a PR for each command.

Make a comment if you wish to tackle this, so I can assign this issue to you.

### PR DESCRIPTION
**Pull request by jason89521**
## Description

Support `no-ignore` option.

## Additional information

Close #54 when this pr is merged.

## Related issue

#54

### CODE REVIEW

File: crates/oxc_cli/src/lib.rs [hunk 1]
```diff
@@ -42,6 +42,10 @@ impl Cli {
                 Walk::new(path, &self.cli_options)
                     .iter()
                     .filter(|path| {
```
Comment (Boshen) [inline crates/oxc_cli/src/lib.rs:44]: I think all these ignore logic should all be moved to the walker, it's unrelated to this linter run anymore.

Comment (Boshen) [reply crates/oxc_cli/src/lib.rs:44]: https://docs.rs/ignore/0.4.20/ignore/struct.WalkBuilder.html#method.filter_entry
Does this work?

Comment (jason89521) [reply crates/oxc_cli/src/lib.rs:44]: The method needs the reference be `static`.
image

Comment (jason89521) [reply crates/oxc_cli/src/lib.rs:44]: I tried 
```rust
		let ignore_pattern = options.ignore_pattern.to_owned();
        if !options.no_ignore {
            inner.add_custom_ignore_filename(&options.ignore_path).filter_entry(move |dir_entry| {
                for pattern in ignore_pattern.iter() {
                    if pattern.matches_path(dir_entry.path()) {
                        return false;
                    }
                }
                true
            });
        }
```

But the output of `cargo run --bin oxc_cli lint --ignore-pattern "*.js"  test.js` still printed the lint result

Comment (jason89521) [reply crates/oxc_cli/src/lib.rs:44]: It seems that the `filter_entry` didn't call the closure. I write a `println!` in it, but it didn't print anything.

Comment (jason89521) [reply crates/oxc_cli/src/lib.rs:44]: I have no idea where the passed in `filter` will be called in [the source code](https://github.com/BurntSushi/ripgrep/blob/master/crates/ignore/src/walk.rs) 🥲

Comment (Boshen) [reply crates/oxc_cli/src/lib.rs:44]: Let's figure it out later then 😄

Comment (jason89521) [reply crates/oxc_cli/src/lib.rs:44]: I found that the `filter` will be called in the [`skip_entry`](https://github.com/BurntSushi/ripgrep/blob/master/crates/ignore/src/walk.rs#L962), and this method is called in [here](https://github.com/BurntSushi/ripgrep/blob/master/crates/ignore/src/walk.rs#L1009-L1029C55). The `filter` closure should be called when we used the `Walk`, but it didn't. 🤔

Comment (Boshen) [reply crates/oxc_cli/src/lib.rs:44]: Let's give up for now 😆 This `ignore` interface seems way too complicated

Comment (jason89521) [reply crates/oxc_cli/src/lib.rs:44]: Okay! If we don't use the `skip_entry` but still want to handle all ignore logic in `walk.rs`, we can store the ignore related variable in `Walk` struct. Then deal with them in the `iter` method. Would you like me do that? Or you have another idea.

Comment (Boshen) [reply crates/oxc_cli/src/lib.rs:44]: Let's differ it, I want to look at why `skip_entry` doesn't work

**Pull request merged**